--- skk.c.orig	2017-08-14 09:07:27.000000000 +0900
+++ skk.c	2021-03-14 17:55:00.417717316 +0900
@@ -1272,17 +1272,17 @@
 }
 
 static char *wide_num_list[] =
-  {"£°", "£±", "£²", "£³", "£´", "£µ", "£¶", "£·", "£¸", "£¹"};
+  {"ï¼", "ï¼‘", "ï¼’", "ï¼“", "ï¼”", "ï¼•", "ï¼–", "ï¼—", "ï¼˜", "ï¼™"};
 static char *kanji_num_list[] =
-  {"¡»", "°ì", "Æó", "»°", "»Í", "¸Ş", "Ï»", "¼·", "È¬", "¶å"};
+  {"ã€‡", "ä¸€", "äºŒ", "ä¸‰", "å››", "äº”", "å…­", "ä¸ƒ", "å…«", "ä¹"};
 static char *kanji_num_position_list[] =
-  {NULL, "½½", "É´", "Àé", "Ëü", NULL, NULL, NULL, "²¯", NULL,
-   NULL, NULL, "Ãû", NULL, NULL, NULL, "µş", NULL, NULL, NULL};
+  {NULL, "å", "ç™¾", "åƒ", "ä¸‡", NULL, NULL, NULL, "å„„", NULL,
+   NULL, NULL, "å…†", NULL, NULL, NULL, "äº¬", NULL, NULL, NULL};
 static char *kanji_check_num_list[] =
-  {"¡»", "°í", "Ğ±", "»²", "»Í", "¸à", "Ï»", "¼·", "È¬", "¶å"};
+  {"ã€‡", "å£±", "å¼", "å‚", "å››", "ä¼", "å…­", "ä¸ƒ", "å…«", "ä¹"};
 static char *kanji_check_num_position_list[] =
-  {NULL, "½¦", "É´", "ïô", "èß", NULL, NULL, NULL, "²¯", NULL,
-   NULL, NULL, "Ãû", NULL, NULL, NULL, "µş", NULL, NULL, NULL};
+  {NULL, "æ‹¾", "ç™¾", "é˜¡", "è¬", NULL, NULL, NULL, "å„„", NULL,
+   NULL, NULL, "å…†", NULL, NULL, NULL, "äº¬", NULL, NULL, NULL};
 
 static char *
 numeric_wide_or_kanji_conv(const char *numstr, int method)
@@ -1291,15 +1291,15 @@
   int i, len;
 
   len = strlen(numstr);
-  mbstr = uim_malloc(len * 2 + 1);
+  mbstr = uim_malloc(len * 3 + 1);
 
   for (i = 0; i < len; i++) {
     if (method == 1)
-      strcpy(&mbstr[i * 2], wide_num_list[numstr[i] - '0']);
+      strcpy(&mbstr[i * 3], wide_num_list[numstr[i] - '0']);
     else
-      strcpy(&mbstr[i * 2], kanji_num_list[numstr[i] - '0']);
+      strcpy(&mbstr[i * 3], kanji_num_list[numstr[i] - '0']);
   }
-  mbstr[len * 2] = '\0';
+  mbstr[len * 3] = '\0';
 
   return mbstr;
 }
@@ -1316,19 +1316,19 @@
   if (len > 20) /* too big number */
     return uim_strdup(numstr);
 
-  mbstr = uim_malloc(len * 2 + 1);
-  mblen = len * 2;
+  mbstr = uim_malloc(len * 3 + 1);
+  mblen = len * 3;
 
   for (i = 0, j = 0; j < len; i++, j++) {
     position = len - j - 1;
     if (numstr[j] == '0') {
       i--;
-      mblen -= 2;
+      mblen -= 3;
       /* check zero at the head */
       if (j == 0) {
 	head_is_zero = 1;
       } else {
-	/* add Ëü, ²¯, Ãû, µş for zero */
+	/* add ä¸‡, å„„, å…†, äº¬ for zero */
 	if ((position >= 4) && ((position % 4) == 0) && !head_is_zero) {
 	  int use_position = 0;
 	  if (j >= 3) {
@@ -1344,10 +1344,10 @@
 	  }
 	  if (use_position) {
 	    i++;
-	    mblen += 2;
-	    if (mblen > len * 2)
-	      mbstr = uim_realloc(mbstr, mblen + 2);
-	    strcpy(&mbstr[i * 2], kanji_num_position_list[position]);
+	    mblen += 3;
+	    if (mblen > len * 3)
+	      mbstr = uim_realloc(mbstr, mblen + 3);
+	    strcpy(&mbstr[i * 3], kanji_num_position_list[position]);
 	  }
 	}
       }
@@ -1358,8 +1358,8 @@
       /* replace numstr[j] with kanji number */
       if (numstr[j] == '1') {
 	/*
-	 * use "°ì" only for the one at the place of °ì, Ëü, ²¯, Ãû,
-	 * µş or °ìÀéËü
+	 * use "ä¸€" only for the one at the place of ä¸€, ä¸‡, å„„, å…†,
+	 * äº¬ or ä¸€åƒä¸‡
 	 */
 	if (((position % 4) == 0) ||
 	    ((position >= 7) &&
@@ -1367,33 +1367,33 @@
 	     (numstr[j + 1] == '0') &&
 	     (numstr[j + 2] == '0') &&
 	     (numstr[j + 3] == '0'))) {
-	  strcpy(&mbstr[i * 2], kanji_num_list[1]);
+	  strcpy(&mbstr[i * 3], kanji_num_list[1]);
 	} else {
 	  i--;
-	  mblen -= 2;
+	  mblen -= 3;
 	}
       } else {
-	strcpy(&mbstr[i * 2], kanji_num_list[numstr[j] - '0']);
+	strcpy(&mbstr[i * 3], kanji_num_list[numstr[j] - '0']);
       }
 
-      /* add ½½, É´, Àé for number whose place is exceeded Ëü */
+      /* add å, ç™¾, åƒ for number whose place is exceeded ä¸‡ */
       if (position > 4) {
 	if ((position % 4) != 0) {
 	  i++;
-	  mblen += 2;
-	  if (mblen > len * 2)
-	    mbstr = uim_realloc(mbstr, mblen + 2);
-	  strcpy(&mbstr[i * 2], kanji_num_position_list[position % 4]);
+	  mblen += 3;
+	  if (mblen > len * 3)
+	    mbstr = uim_realloc(mbstr, mblen + 3);
+	  strcpy(&mbstr[i * 3], kanji_num_position_list[position % 4]);
 	}
       }
 
       /* add position */
       if (kanji_num_position_list[position]) {
 	i++;
-	mblen += 2;
-	if (mblen > len * 2)
-	  mbstr = uim_realloc(mbstr, mblen + 2);
-	strcpy(&mbstr[i * 2], kanji_num_position_list[position]);
+	mblen += 3;
+	if (mblen > len * 3)
+	  mbstr = uim_realloc(mbstr, mblen + 3);
+	strcpy(&mbstr[i * 3], kanji_num_position_list[position]);
       }
     }
   }
@@ -1401,7 +1401,7 @@
   /* in case of zero */
   if (head_is_zero) {
     strcpy(&mbstr[0], kanji_num_list[0]);
-    mblen = 2;
+    mblen = 3;
   }
 
   mbstr[mblen] = '\0';
@@ -1420,19 +1420,19 @@
   if (len > 20) /* too big number */
     return uim_strdup(numstr);
 
-  mbstr = uim_malloc(len * 2 + 1);
-  mblen = len * 2;
+  mbstr = uim_malloc(len * 3 + 1);
+  mblen = len * 3;
 
   for (i = 0, j = 0; j < len; i++, j++) {
     position = len - j - 1;
     if (numstr[j] == '0') {
       i--;
-      mblen -= 2;
+      mblen -= 3;
       /* check zero at the head */
       if (j == 0) {
 	head_is_zero = 1;
       } else {
-	/* add èß, ²¯, Ãû, µş for zero */
+	/* add è¬, å„„, å…†, äº¬ for zero */
 	if ((position >= 4) && ((position % 4) == 0) && !head_is_zero) {
 	  int use_position = 0;
 	  if (j >= 3) {
@@ -1448,10 +1448,10 @@
 	  }
 	  if (use_position) {
 	    i++;
-	    mblen += 2;
-	    if (mblen > len * 2)
-	      mbstr = uim_realloc(mbstr, mblen + 2);
-	    strcpy(&mbstr[i * 2], kanji_check_num_position_list[position]);
+	    mblen += 3;
+	    if (mblen > len * 3)
+	      mbstr = uim_realloc(mbstr, mblen + 3);
+	    strcpy(&mbstr[i * 3], kanji_check_num_position_list[position]);
 	  }
 	}
       }
@@ -1460,26 +1460,26 @@
 	head_is_zero = 0;
 
       /* replace numstr[j] with kanji number */
-      strcpy(&mbstr[i * 2], kanji_check_num_list[numstr[j] - '0']);
+      strcpy(&mbstr[i * 3], kanji_check_num_list[numstr[j] - '0']);
 
-      /* add ½¦, É´, ïô for number whose place is exceeded èß */
+      /* add æ‹¾, ç™¾, é˜¡ for number whose place is exceeded è¬ */
       if (position > 4) {
 	if ((position % 4) != 0) {
 	  i++;
-	  mblen += 2;
-	  if (mblen > len * 2)
-	    mbstr = uim_realloc(mbstr, mblen + 2);
-	  strcpy(&mbstr[i * 2], kanji_check_num_position_list[position % 4]);
+	  mblen += 3;
+	  if (mblen > len * 3)
+	    mbstr = uim_realloc(mbstr, mblen + 3);
+	  strcpy(&mbstr[i * 3], kanji_check_num_position_list[position % 4]);
 	}
       }
 
       /* add position */
       if (kanji_check_num_position_list[position]) {
 	i++;
-	mblen += 2;
-	if (mblen > len * 2)
-	  mbstr = uim_realloc(mbstr, mblen + 2);
-	strcpy(&mbstr[i * 2], kanji_check_num_position_list[position]);
+	mblen += 3;
+	if (mblen > len * 3)
+	  mbstr = uim_realloc(mbstr, mblen + 3);
+	strcpy(&mbstr[i * 3], kanji_check_num_position_list[position]);
       }
     }
   }
@@ -1487,7 +1487,7 @@
   /* in case of zero */
   if (head_is_zero) {
     strcpy(&mbstr[0], kanji_check_num_list[0]);
-    mblen = 2;
+    mblen = 3;
   }
 
   mbstr[mblen] = '\0';
@@ -1504,10 +1504,10 @@
   if (len != 2) /* allow two digit number only */
     return uim_strdup(numstr);
 
-  mbstr = uim_malloc(5);
+  mbstr = uim_malloc(7);
   strcpy(&mbstr[0], wide_num_list[numstr[0] - '0']);
-  strcpy(&mbstr[2], kanji_num_list[numstr[1] - '0']);
-  mbstr[4] = '\0';
+  strcpy(&mbstr[3], kanji_num_list[numstr[1] - '0']);
+  mbstr[6] = '\0';
 
   return mbstr;
 }
@@ -1525,17 +1525,17 @@
   case 0:
     ret = uim_strdup(numstr);
     break;
-  case 1: /* Á´³Ñ¿ô»ú */
-  case 2: /* ´Á¿ô»ú °Ì¼è¤êÌµ¤· */
+  case 1: /* å…¨è§’æ•°å­— */
+  case 2: /* æ¼¢æ•°å­— ä½å–ã‚Šç„¡ã— */
     ret = numeric_wide_or_kanji_conv(numstr, method);
     break;
-  case 3: /* ´Á¿ô»ú °Ì¼è¤êÍ­¤ê */
+  case 3: /* æ¼¢æ•°å­— ä½å–ã‚Šæœ‰ã‚Š */
     ret = numeric_kanji_with_position_conv(numstr);
     break;
-  case 5: /* ¾®ÀÚ¼êÉ½µ­ */
+  case 5: /* å°åˆ‡æ‰‹è¡¨è¨˜ */
     ret = numeric_kanji_for_check_conv(numstr);
     break;
-  case 9: /* ¾­´ıÉ½µ­ */
+  case 9: /* å°†æ£‹è¡¨è¨˜ */
     ret = numeric_shogi_conv(numstr);
     break;
   default:
